allensdk/test/brain_observatory/behavior/test_stimulus_processing.py

```diff
@@ -1,52 +1,141 @@
-
+import pandas as pd
 import numpy as np
 import pytest
 
-
 from allensdk.brain_observatory.behavior.stimulus_processing import (
-    get_stimulus_presentations, _get_stimulus_epoch, _get_draw_epochs)
-
-
-data = {
-    "items": {
-        "behavior": {
-            "stimuli": {
-                "images": {
-                    "set_log": [
-                        ('Image', 'im065', 5.809955710916157, 0),
-                        ('Image', 'im061', 314.06612555068784, 6),
-                        ('Image', 'im062', 348.5941232265203, 12),
-                    ],
-                    "draw_log": ([0]+[1]*3 + [0]*3)*3 + [0]
-                }
-            },
-            # "intervalsms": np.array([16.0]*10)
+    get_stimulus_presentations, _get_stimulus_epoch, _get_draw_epochs,
+    get_visual_stimuli_df, get_stimulus_metadata)
+
+
+@pytest.fixture()
+def behavior_stimuli_time_fixture(request):
+    """
+    Fixture that allows for parameterization of behavior_stimuli stimuli
+    time data.
+    """
+    timestamp_count = request.param.get("timestamp_count", 22)
+    time_step = request.param.get("time_step", 0.016)
+
+    fixture_params = {
+        "timestamp_count": timestamp_count,
+        "time_step": time_step
+    }
+
+    timestamps = np.array([time_step * i for i in range(timestamp_count)])
+
+    return timestamps, fixture_params
+
+
+@pytest.fixture()
+def behavior_stimuli_data_fixture(request):
+    """
+    This fixture mimicks the behavior experiment stimuli data logs and
+    allows parameterization for testing
+    """
+    images_set_log = request.param.get("images_set_log", [
+        ('Image', 'im065', 5.809, 0)])
+    images_draw_log = request.param.get("images_draw_log", [
+        ([0] + [1]*3 + [0]*3)
+    ])
+    grating_set_log = request.param.get("grating_set_log", [
+        ('Ori', 90, 3.585, 0)
+    ])
+    grating_draw_log = request.param.get("grating_draw_log", [
+        ([0] + [1]*3 + [0]*3)
+    ])
+    omitted_flash_frame_log = request.param.get("omitted_flash_frame_log", {
+        "grating_0": []
+    })
+    grating_phase = request.param.get("grating_phase", None)
+    grating_correct_frequency = request.param.get("grating_correct_frequency",
+                                                  None)
+
+    data = {
```

Comment:
Is using this fixture in fact any more succinct than just using this structure as the parameter?

Reply:
I think it's more succinct, I don't want to nest the data four levels down in each tests case as I thought that was less readable.